### DISCUSSION TITLE
Example of moor_flutter

### DISCUSSION THREAD

**Discussion by kernju** [simolus3/drift#1474] [Q&A]
Hi,

It seems that the example on moor_flutter is no longer working (the one at moor/moor_flutter/example/ and the moor/moor_flutter/example/lib/database/database.dart file in particular). Do you confirm ? Is there any other example code working ?

Here is the list of the errors I get :
![image](https://user-images.githubusercontent.com/81294413/136224112-99a86876-2855-4e90-80f8-12c8543254ba.png)

It seems there is a lot with the null-safety mechanism.

**Comment by simolus3** [comment]
Thanks for pointing this out! Which version of `moor_flutter` did you check out? The current one on `develop` uses an old project template with deprecations in native code, but the Dart code is fine and compiles with Flutter 2.5 and the latest dependencies (albeit without null safety). The project has a `2.6.0` sdk environment in its pubspec, so null-safety related warnings should not apply :)

**Reply by kernju** [reply]
You are welcome :)
In fact I just picked out some of the files and did not check out properly.
But this time I checked out the entire repo, and getting this when running `flutter pub get` in the `moor_flutter/example/` directory : 
```
C:\flutter\bin\flutter.bat --no-color pub get
Running "flutter pub get" in example...                         
Warning: You are using these overridden dependencies:
! moor 4.6.0-dev from path ..\..\moor
! moor_flutter 4.1.0-dev from path ..
! moor_generator 4.5.1 from path ..\..\moor_generator
! sqlparser 0.17.2 from path ..\..\sqlparser
Running "flutter pub get" in example...                          2 943ms
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Warning
──────────────────────────────────────────────────────────────────────────────
Your Flutter application is created using an older version of the Android
embedding. It's being deprecated in favor of Android embedding v2. Follow the
steps at

https://flutter.dev/go/android-project-migration

to migrate your project.
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

This app is using a deprecated version of the Android embedding.
To avoid unexpected runtime failures, or future build failures, try to migrate this app to the V2 embedding.
Take a look at the docs for migrating an app: https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects
```

Maybe there are some tips I am not aware of regarding the null-safety ?

**Reply by simolus3** [reply]
You're write, the original example project is quite old and is still using the v1 android embedding. I have migrated it to null safety and all the latest dependencies in dcfddcbb3b38de878ac708fb65a4071cd4011a2d though.
